### DISCUSSION TITLE
Possible to restrict new users to View Only?

### DISCUSSION THREAD

**Discussion by chase-replicated** [outline/outline#2130] [Self Hosting] [answered]
We're considering allowing users to sign into our company wiki as viewers so they can read our company handbook and other public facing documents. One thing blocking this is the fact that new users by default are able to create new collections and edit any collections that have default access set to view and edit. 

This is problematic as it allows for bad actors if we allow users into our wiki. One solution is to set up a tree of publicly shared documents that together constitute the internet-facing documents, but we're challenged to be as open as possible, so allowing users into the app with a default role of "Viewer" would be preferred. Additionally, If we could publish "Collections" to be public and not just "Documents" that would also likely fulfill the use case.

I saw this discussion: https://github.com/outline/outline/discussions/1662 which proposes a SQL trigger for adding new users to a specified group, but I'm not sure how that would be implemented entirely or if it would solve the problem I'm facing.

**Comment by tommoor** [comment, marked as answer]
> One solution is to set up a tree of publicly shared documents that together constitute the internet-facing documents

This work is almost complete and will be in the next release: https://github.com/outline/outline/pull/2075 – there's also future work on the medium-term roadmap to publish an entire collection as a website.

> allowing users into the app with a default role of "Viewer" would be preferred

The ability to set a "default role" in the team security settings that would be applied to newly created users is something that we'd accept a PR for. We have read-only users currently, there's just no way to say that all new users should get this role by default.

**Reply by chase-replicated** [reply]
Awesome, thanks for the great app! I'll look into the default role when I've got some time.

**Reply by Zilborg** [reply]
Does something change for new users? I can't find configuration for set default role = 'Viewer' for new users.

**Reply by tommoor** [reply]
Hi @Zilborg default role functionality has not yet been built unfortunately, I believe it would be a very simple feature though.
